lms/static/scripts/frontend_apps/utils/api.ts

```diff
@@ -315,6 +315,14 @@ export type PaginatedFetchResult<T> = Omit<FetchResult<T>, 'mutate'> & {
   /** Determines if currently loading the first page */
   isLoadingFirstPage: boolean;
   loadNextPage: () => void;
+
+  /**
+   * Determines the size of every page (maximum number of results returned per
+   * page).
+   * If the API indicates there is only one page, this will be undefined. You
+   * can trust `data.length` as the total number of items in that case.
+   */
+  pageSize?: number;
```

Comment:
For the use case described in the issue, I think it would be sufficient to have a boolean to indicate whether there are more pages to fetch or not. Then in the count badges you can display just the number if `data.length` is the total or `` `${data.length}`+`` otherwise.

This would mean that if the user opened the list and scrolled enough to trigger fetching additional pages, the badge count would change to reflect the new lower bound on the result size. I'm not sure if we want that behavior or not.